src/server/worker.cc

```diff
@@ -430,6 +430,10 @@ void Worker::KillClient(Redis::Connection *self, uint64_t id, std::string addr,
   for (const auto &iter : conns_) {
     Redis::Connection *conn = iter.second;
     if (skipme && self == conn) continue;
+    // no need to kill again the client if flags as kCloseAfterReply
```

Comment:
So we kill clients asynchronously?

Reply:
Yes, it will close the client after the socket buffer were sent out.